packages/next-swc/crates/core/src/relay.rs

```diff
@@ -0,0 +1,169 @@
+                        || word == "fragment"
+                    {
+                        return Some(next_word);
+                    }
+
+                    None
+                })
+                .next();
+
+            return operation;
+        })
+        .next()
+}
+
+fn build_require_expr_from_path(path: String) -> Expr {
+    Expr::Call(CallExpr {
+        span: Default::default(),
+        callee: Callee::Expr(Box::new(Expr::Ident(Ident::new(
+            js_word!("require"),
+            Span::default(),
+        )))),
+        args: vec![ExprOrSpread {
+            spread: None,
+            expr: Box::new(Expr::Lit(Lit::Str(Str {
+                span: Default::default(),
+                value: JsWord::from(path),
+                has_escape: false,
+                kind: Default::default(),
+            }))),
+        }],
+        type_args: None,
+    })
+}
+
+impl Fold for Relay {
+    fn fold_expr(&mut self, expr: Expr) -> Expr {
+        let expr = expr.fold_children_with(self);
+
+        match &expr {
+            Expr::TaggedTpl(tpl) => {
+                if let Some(built_expr) = self.build_call_expr_from_tpl(tpl) {
+                    built_expr
+                } else {
+                    expr
+                }
+            }
+            _ => expr,
+        }
+    }
+}
+
+// This is copied from https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/src/build_project/generate_artifacts.rs#L251
+// until the Relay team exposes it for external use.
+fn path_for_artifact(project_config: &ProjectConfig, definition_name: &str) -> PathBuf {
+    let source_file_location_key = SourceLocationKey::Standalone {
+        // TODO: Figure out why passing in the actual path here causes the requires to be an
+        // absolute path.
+        path: StringKey::from_str("NA").unwrap(),
+    };
+    let filename = if let Some(filename_for_artifact) = &project_config.filename_for_artifact {
+        filename_for_artifact(
+            source_file_location_key,
+            StringKey::from_str(definition_name).unwrap(),
+        )
+    } else {
+        match &project_config.typegen_config.language {
+            relay_config::TypegenLanguage::Flow => format!("{}.graphql.js", definition_name),
+            relay_config::TypegenLanguage::TypeScript => format!("{}.graphql.ts", definition_name),
+        }
+    };
+    create_path_for_artifact(project_config, source_file_location_key, filename)
+}
+
+impl Relay {
+    fn build_require_path(&mut self, operation_name: &str) -> PathBuf {
+        match &self.relay_config_for_tests {
+            Some(config) => path_for_artifact(config, operation_name),
+            _ => {
+                let config = relay_compiler::config::Config::search(
+                    std::env::current_dir().unwrap().as_path(),
+                )
+                .unwrap();
+
+                let project_config = &config.projects[&StringKey::from_str("default").unwrap()];
```

Comment:
@alunyov I'm just pulling the default project config here since we're assuming they're using a `SingleProjectConfig`. 

@kdy1 Can you point me to some code on how we should handle errors? I would like to spit out some user friendly errors in a few cases.

Reply:
@kdy1 I think Rust is having a hard time inferring the type there 
```
error[E0284]: type annotations needed
   --> crates/core/src/relay.rs:122:66
    |
122 |                 let project_config = &config.projects[&"default".parse().unwrap()];
    |                                                                  ^^^^^ cannot infer type for type parameter `F` declared on the associated function `parse`
```

Reply:
You can use qualified syntax like `.parse::<Type>().unwrap()`.